sql/hive/src/main/scala/org/apache/spark/sql/hive/HiveContext.scala

```diff
@@ -271,11 +271,15 @@ class HiveContext(sc: SparkContext) extends SQLContext(sc) {
    * @since 1.3.0
    */
   def refreshTable(tableName: String): Unit = {
-    catalog.refreshTable(catalog.client.currentDatabase, tableName)
+    val dbAndTableName = tableName.split("\\.")
+    catalog.refreshTable(dbAndTableName.lift(dbAndTableName.size -2)
+      .getOrElse(catalog.client.currentDatabase), dbAndTableName.last)
   }
 
   protected[hive] def invalidateTable(tableName: String): Unit = {
-    catalog.invalidateTable(catalog.client.currentDatabase, tableName)
+    val dbAndTableName = tableName.split("\\.")
```

Comment:
notice repeated code.
IMHO relative/fully-prefixed tableName logic should be extracted into something like `cannonicalTableName(relativeOrPrefixedTableName)`